components/locid/src/subtags/region.rs

```diff
@@ -2,160 +2,41 @@
-    ///
-    /// assert_eq!(region, "FR");
-    /// ```
-    pub const fn from_bytes(v: &[u8]) -> Result<Self, ParserError> {
-        Self::from_bytes_manual_slice(v, 0, v.len())
-    }
-
-    /// Equivalent to [`from_bytes(bytes[start..end])`](Self::from_bytes),
-    /// but callable in a `const` context (which range indexing is not).
-    pub const fn from_bytes_manual_slice(
-        v: &[u8],
-        start: usize,
-        end: usize,
-    ) -> Result<Self, ParserError> {
-        match end - start {
-            REGION_ALPHA_LENGTH => match TinyAsciiStr::from_bytes_manual_slice(v, start, end) {
-                Ok(s) if s.is_ascii_alphabetic() => Ok(Self(s.to_ascii_uppercase())),
-                _ => Err(ParserError::InvalidSubtag),
-            },
-            REGION_NUM_LENGTH => match TinyAsciiStr::from_bytes_manual_slice(v, start, end) {
-                Ok(s) if s.is_ascii_numeric() => Ok(Self(s)),
-                _ => Err(ParserError::InvalidSubtag),
-            },
-            _ => Err(ParserError::InvalidSubtag),
-        }
-    }
-
-    /// Safely creates a [`Region`] from a reference to its raw format
-    /// as returned by [`Region::into_raw()`].
-    ///
-    /// # Examples
-    ///
-    /// ```
-    /// use icu::locid::subtags::Region;
-    ///
-    /// assert!(matches!(Region::try_from_raw(*b"US\0"), Ok(_)));
-    /// assert!(matches!(Region::try_from_raw(*b"419"), Ok(_)));
-    /// assert!(matches!(Region::try_from_raw(*b"foo"), Err(_)));
-    /// assert!(matches!(Region::try_from_raw(*b"US0"), Err(_)));
-    ///
-    /// // Unlike the other constructors, this one is case-sensitive:
-    /// assert!(matches!(Region::try_from_raw(*b"us\0"), Err(_)));
-    /// ```
-    pub fn try_from_raw(v: [u8; 3]) -> Result<Self, ParserError> {
-        let s = TinyAsciiStr::<{ core::mem::size_of::<Self>() }>::try_from_raw(v)
-            .map_err(|_| ParserError::InvalidSubtag)?;
-        let is_valid = match s.len() {
-            REGION_ALPHA_LENGTH => s.is_ascii_alphabetic_uppercase(),
-            REGION_NUM_LENGTH => s.is_ascii_numeric(),
-            _ => false,
-        };
-        if is_valid {
-            Ok(Self(s))
-        } else {
-            Err(ParserError::InvalidSubtag)
-        }
-    }
-
-    /// Deconstructs the [`Region`] into raw format to be consumed
-    /// by [`from_raw_unchecked()`](Region::from_raw_unchecked()).
-    ///
-    /// # Examples
-    ///
-    /// ```
-    /// use icu::locid::subtags::Region;
-    ///
-    /// let region = Region::from_bytes(b"us").expect("Parsing failed.");
-    ///
-    /// let raw = region.into_raw();
-    /// let region = unsafe { Region::from_raw_unchecked(raw) };
-    /// assert_eq!(region, "US");
-    /// ```
-    pub fn into_raw(self) -> [u8; 3] {
-        *self.0.all_bytes()
-    }
+#[inline]
+const fn validate(s: &TinyAsciiStr<3>) -> bool {
+    (s.len() != 2 || s.is_ascii_alphabetic()) && (s.len() != 3 || s.is_ascii_numeric())
```

Comment:
`nit`: it's much easier for humans to read positive than negative statements. Can you flip the logic here to `==`+`&&`?

Reply:
It's even easier to read if/else so I did that.